lib/traject/bibtex_config.rb

```diff
@@ -17,71 +20,26 @@
   context.skip!("Skipping #{record.key} no keywords") unless record.respond_to?(:keywords)
 end
 
-to_field 'id', from_bibtex(:id)
-
-to_field 'bibtex_key_ss', from_bibtex(:key)
-
-to_field 'ref_type_ssm', from_bibtex(:ref_type)
-
-to_field 'title_display', from_bibtex(:title)
-to_field 'title_full_display', from_bibtex(:title)
-to_field 'title_uniform_search', from_bibtex(:title)
-to_field 'title_sort', from_bibtex(:title)
-
-to_field 'author_person_full_display', from_bibtex(:author)
-to_field 'author_sort', from_bibtex(:author)
-
-to_field 'pub_year_isi', from_bibtex(:year)
-to_field 'pub_year_w_approx_isi', from_bibtex(:year)
-
-to_field 'editor_ssim', from_bibtex(:editor)
-
-to_field 'book_title_ssim', from_bibtex(:booktitle)
-
-to_field 'pub_display', lambda { |record, accumulator, _context|
-  accumulator << record.journal.to_s.presence if record.respond_to?(:journal)
-  accumulator << record.publisher.to_s.presence if record.respond_to?(:publisher)
-}
-
-to_field 'location_ssi', from_bibtex(:address)
-
-to_field 'university_ssim', from_bibtex(:school)
-
-to_field 'edition_ssm', from_bibtex(:edition)
-
-to_field 'series_ssi', from_bibtex(:series)
-
-to_field 'thesis_type_ssm', from_bibtex(:type)
-
-to_field 'volume_ssm', from_bibtex(:volume)
-
-to_field 'issue_ssm', from_bibtex(:issue)
-
-to_field 'pages_ssm', from_bibtex(:pages)
-
-to_field 'doi_ssim', from_bibtex(:doi)
-
-to_field 'general_notes_ssim', lambda { |record, accumulator, _context|
-  accumulator << record.annote.to_s.presence if record.respond_to?(:annote)
-}
-
+to_field 'id', extract_bibtex_id
+to_field 'bibtex_key_ss', extract_bibtex_key
+to_field 'ref_type_ssm', extract_bibtex_field(:ref_type)
+to_fields %w(title_display title_full_display title_uniform_search title_sort), extract_bibtex_field(:title)
+to_fields %w(author_person_full_display author_sort), extract_bibtex_field(:author)
+to_fields %w(pub_year_isi pub_year_w_approx_isi), extract_bibtex_field(:year)
+to_field 'editor_ssim', extract_bibtex_field(:editor)
+to_field 'book_title_ssim', extract_bibtex_field(:booktitle)
+to_field 'pub_display', extract_bibtex_publication
+to_field 'location_ssi', extract_bibtex_field(:address)
+to_field 'university_ssim', extract_bibtex_field(:school)
+to_field 'edition_ssm', extract_bibtex_field(:edition)
+to_field 'series_ssi', extract_bibtex_field(:series)
+to_field 'thesis_type_ssm', extract_bibtex_field(:type)
+to_field 'volume_ssm', extract_bibtex_field(:volume)
+to_field 'issue_ssm', extract_bibtex_field(:issue)
+to_field 'pages_ssm', extract_bibtex_field(:pages)
+to_field 'doi_ssim', extract_bibtex_field(:doi)
+to_field 'general_notes_ssim', extract_bibtex_field(:annote)
```

Comment:
Same here as with the changes requested below - once extraction macro support is setup for `extract_bibtext_field`, and annotations are concatenated following whatever character, this can become: 

```
to_field 'general_notes_ssim', extract_bibtex('annote', split: '|', trim: true)
```

discussed briefly with @camillevilla

Reply:
ok, i'm keeping this little bit of extraction logic in bib_config for now.